src/Microsoft.AspNetCore.Blazor.Razor.Extensions/BindTagHelperDescriptorProvider.cs

```diff
@@ -0,0 +1,491 @@
+            //
+            // We handle a few different cases here:
+            //
+            //  1.  When given an attribute like **anywhere**'bind-value-changed="@FirstName"' we will
+            //      generate the 'value' attribute and 'changed' attribute. 
+            //
+            //      We don't do any transformation or inference for this case, because the developer has
+            //      told us exactly what to do. This is the *full* form of bind, and should support any
+            //      combination of elemement, component, and attributes.
+            //
+            //      This is the most general case, and is implemented with a built-in tag helper that applies
+            //      to everything, and binds to a dictionary of attributes that start with bind-.
+            //
+            //  2.  We also support cases like 'bind-value="@FirstName"' where we will generate the 'value'
+            //      attribute and and another attribute based for a changed handler based on the metadata.
+            //
+            //     These mappings are provided by attributes that tell us what attributes, suffixes, and
+            //      elements to map.
+            //
+            //  3.  When given an attribute like 'bind="@FirstName"' we will generate a value and change
+            //      attribute solely based on the context. We need the context of an HTML tag to know
+            //      what attributes to generate.
+            //
+            //      Similar to case #2, this should 'just work' from the users point of view. We expect
+            //      using this syntax most frequently with input elements.
+            //
+            //      These mappings are also provided by attributes. Primarily these are used by <input />
+            //      and so we have a special case for input elements and their type attributes.
+            //
+            //  4.  For components, we have a bit of a special case. We can infer a syntax that matches
+            //      case #2 based on property names. So if a component provides both 'Value' and 'ValueChanged'
+            //      we will turn that into an instance of bind.
+            //
+            // So case #1 here is the most general case. Case #2 and #3 are data-driven based on attribute data
+            // we have. Case #4 is data-driven based on component definitions.
+            //
+            // We provide a good set of attributes that map to the HTML dom. This set is user extensible.
+            var compilation = context.GetCompilation();
+            if (compilation == null)
+            {
+                return;
+            }
+
+            var bindMethods = compilation.GetTypeByMetadataName(BlazorApi.BindMethods.FullTypeName);
+            if (bindMethods == null)
+            {
+                // If we can't find BindMethods, then just bail. We won't be able to compile the
+                // generated code anyway.
+                return;
```

Comment:
We don't throw because this happens in the compiler. 

The case where this would happen in unresolved references. If you don't have a reference to `Microsoft.AspNet.Blazor`, you'll know :)